app/src/pages/inside/stepPage/modals/editDefectModal/editDefectModal.jsx

```diff
@@ -207,14 +171,62 @@ export class EditDefectModal extends Component {
     return getStorageItem(REPLACE_COMMENTS_TO_ALL_AVAILABILITY_NAME);
   };
 
+  getCloseConfirmationConfig = () =>
+    this.checkIfTheDataWasChanged()
+      ? {
+          confirmationWarning: this.props.intl.formatMessage(messages.warningMessage),
+        }
+      : null;
+
+  getItemsToNextAction = () =>
+    this.mapStateToFetchData().map((item) => ({ ...item, id: item.test_item_id }));
```

Comment:
can adding `id` be done in `mapStateToFetchData`?

Reply:
This method prepare saved data from edit defect modal to another modal (f.e. link, unlink etc.), so we need to get object with the same structure as `props.data.items`, but with part of new data delivered from edit defect modal.